pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -785,6 +785,12 @@ def multi_head_attention_forward(
     return attn_output, None
 
 
+def cosine_similarity(x1: Tensor, x2: Tensor, dim: Optional[int] = 1, eps: Optional[float] = 1e-8) -> Tensor:
```

Comment:
You don't use the `eps`. This is probably an error. It probably is added to the input of `rsqrt`?

Also, is `eps=None` really allowed? Otherwise `Optional` is wrong.

Reply:
It's not added, but the maximum of the norms and `eps` is taken, see https://pytorch.org/docs/stable/generated/torch.nn.CosineSimilarity.html

Reply:
Do you think using `min(cat(...))` makes sense here? Then it has to be fixed for scalars I think.

Reply:
Ok, anyway, we should do the same then.